### ISSUE TITLE
Resolve nested dependencies

### ISSUE THREAD

**Issue by abeMedia** [stormcat24/protodep#25]
Hi @stormcat24, I just realised that dependencies of dependencies don't get resolved. How about checking for the existence of `protodep.toml` after cloning a repo and pulling in any dependencies within the same path?

**Comment by stormcat24** [comment]
@abeMedia That's right. Nested dependencies are unsupported yet. When I made protodep at first, it was not necessary. But this feature will soon be required.

> How about checking for the existence of protodep.toml after cloning a repo and pulling in any dependencies within the same path?

I pretty much agree, but I'm concerned about cyclic dependencies. When protodep scans dependencies recursively, this problem can be avoided by checking for duplicates.

And one more, case of different revisions in duplicated dependencies. Which revision should be choiced? I think shallow hierarchy should be given high priority.

**Comment by abeMedia** [comment]
Hmm good point I hadn't given it that much thought. For now we could simply throw an error about a version conflict with a flag to ignore all conflicts (and take the current repository's revision by default). This might not be ideal behaviour but in my opinion its a good starting point and fairly easy to implement.

As for cyclic imports, that is a tricky one... My feeling is to express a warning in the docs that they are forbidden and trust that all users will be wise enough to listen :rofl:

**Comment by stormcat24** [comment]
> For now we could simply throw an error about a version conflict with a flag to ignore all conflicts (and take the current repository's revision by default).

I think this method is best at now 👍 

Although avoiding cyclic will be a future feature, nest dependency should be implemented.

**Comment by hojongs** [comment]
Is there any update?? This is so nice feature
